### DISCUSSION TITLE
Can we use this to read from Mitsubishi F Series PLC

### DISCUSSION THREAD

**Discussion by Rahul14189** [pymodbus-dev/pymodbus#1760] [Q&A]
Hi, 
I am looking to connect and read from Mitsubishi F Series PLC over the MELSEC protocol, so can I use PYMODBUS to perform the same?

**Comment by janiversen** [comment]
I do not have an idea....if the MELSEC protocol is the same as MODBUS protocol then possibly.

**Reply by Rahul14189** [reply]
As per the log, i am able to establish connection in both TCP and UDP scenario, but fails to read !
Sharing TCP connected read log below:

DEBUG:pymodbus.client.sync:Connection to Modbus server established.
![image](https://github.com/pymodbus-dev/pymodbus/assets/99563180/ebe487d8-1eeb-46ae-a0de-dd0537c852a9)

**Reply by janiversen** [reply]
Modbus protocol for TCP contains a so called MDAP header which is 6 bytes, so a reply of 2 bytes is illegal !

Did you check, whether the device actually uses the modbus protocol and if so, which type of framer it uses (socket or rtu).

I am sorry but this is really not something I can help with, that is something you need to either google or read the manuals, Mitsubishi provides very complete manuals, so its only a matter of finding the right manual.

**Reply by janiversen** [reply]
Why do you think that modbus is the same as MELSEC, did you google it ?

**Reply by Rahul14189** [reply]
Yes, I googled it and found in some blog, that pymodbus library can be used for melsec protocol, so I just gave it a try.
Right now, there is some issue in my mitsubishi Fx3u PLC, after fixing it, i will again give it a try if it works or not, I will update here. 
Thanks @janiversen for the help
